services/data/src/react/hooks/useDataQuery.ts

```diff
@@ -103,7 +104,19 @@ export const useDataQuery = (
         }
 
         if (newVariables) {
-            setVariables({ ...variables, ...newVariables })
+            const merged = { ...variables, ...newVariables }
+            const identical =
+                stableValueHash(variables) === stableValueHash(merged)
```

Comment:
Probably premature optimization but we could technically store the hash of `variables` so it doesn't get re-computed later

Reply:
That sounds like a good optimization. I'd prefer to delay that until after the release and potentially address it in a patch update. I could use a modern version of https://benchmarkjs.com and add a perf test, update the code to include the suggested fix. Then we'd also have clear data on the improvement. Might be a good moment to start testing that workflow (I'm always suggesting we perf test before doing perf updates, might as well start doing that).

Reply:
Say with https://github.com/paularmstrong/react-component-benchmark for example.

Reply:
Ok, so I benchmarked it, and not using a ref to store the computed hash is on average about 2% slower for the test I ran:

<img width="519" alt="Screenshot 2021-08-10 at 15 40 08" src="https://user-images.githubusercontent.com/7355199/128877377-5fc7efac-72e3-4c63-8ec1-bd4c19d45f1c.png">

So seems like a net performance gain. I'll add it.

Reply:
See 3025dc7, let me know if that's what you were thinking of.